src/main/java/com/emc/ecs/nfsclient/nfs/io/NfsFileBase.java

```diff
@@ -77,27 +86,30 @@
      *            The supporting NFS client.
      * @param path
      *            The full path of the file, starting with the mount point.
+     * @throws IOException 
      */
-    public NfsFileBase(N nfs, String path) {
+    public NfsFileBase(N nfs, String path) throws IOException {
         if (nfs == null) {
             throw new IllegalArgumentException("Nfs instance can not be null");
         }
         _nfs = nfs;
-
-        setPath(path);
+        _parent = makeParentPath(path);
```

Comment:
Yes, that means there isn't a parent above the root folder. There is still a path for that folder, though, so that the code to create child paths is the same as for every other file.